richardaspinall/poker-react-node
# feat: user authentication logic for signin endpoint

### Description

This is the meat of the API for signing in a user (minus the session work) 

1. The new `UserService` class has a validatePassword function (obvious what it does). But this service is where we should add other User related operations. The class serves as an abstraction layer between the user data repository and the application's service handlers
2. The `UserRepository` has a new `getPassword` function. This class acts as another abstraction and interacts with the DB for user related operations.
3. I've added another mock function to `dbMocks` which returns a success but no rows (in this case it was for a username that does not exist). 

Note: we should add hashing for passwords but I want that to be a separate specific PR  

### How to test



> Ensure you have set up the DB (see readme)
> Run: `npm run create-dev-db` and then `cd backend && npm run create-test-db` 

1. Head to Sign in
2. Enter `username: richard, password: testpassword`, `thomas, testpassword` or `jon, testpassword`
3. Redirect to `Play`

### Task

[CU-86cuz3dq9 ](https://app.clickup.com/t/86cuz3dq9)

### Checklist
